mpi-proxy-split/doc/nimrod-build-tutorial.txt

```diff
@@ -0,0 +1,273 @@
+
+## SUBSEQUENT BUILDS ##
+
+As best written in the DEVELOPMENT WORKFLOW section of nimall/builds:
+
+  Rebuilding with bilder is not required when a change is made to the NIMROD
+  source files. In fact, rebuilding with bilder is also not recommended for
+  development.  If an svn repo is not clean (e.g. nimall/nimdevel has local
+  modifications) bilder will not install the code unless passed the -F flag. If
+  the -F flag is passed the modified code will be installed with the -rXXXXM tag
+  where XXXX is the svn repo revision number.  The softlinks in the $NIMSOFT
+  (install) directory will be updated to point to this installation and thus the
+  binary placed in the PATH variable when nimall.sh is sourced will be updated.
+  
+  The recommended workflow for development is work directly in the <build
+  directory>/nimdevel/<ser/par> directories with debug builds (see below).
+  Execute
+  
+    $ make
+  
+  to rebuild and incorporate your changes. Use
+  
+    $ make -j4
+  
+  to build in parallel with 4 processors. These commands will produce a modified
+  binaries in the <build directory>/nimdevel/<ser/par>/bin directory. One can
+  install these binaries with
+  
+    $ make install
+  
+  however this is not recommended as it will corrupt the installation directory
+  naming system. NIMROD uses a cmake build system and available build targets
+  are shown with
+  
+    $ make help
+  
+  When new source files are added, the corresponding CMakeLists.txt file must be
+  changed in the source code directory and the new file added to the svn repo.
+
+4. Testing NIMROD
+
+### ON CORI ###
+
+Please ask Zhengji Zhao (zzhao@lbl.gov) for access to the test case on Cori.
+
+To run it, navigate to the given directory. Either submit a batch script or run
+it manually.
+
+  $ salloc -N 11 -C haswell -q interactive -t 00:15:00 # customize yourself
+  $ mana_coordinator # optionally; refer to MANA documentation for installation
+  $ srun \
+      --ntasks=88 \
+      --ntasks-per-node=8 \
+      --cpus-per-task=2 \
+      --cpu-bind=cores \
+      MANA_BIN/mana_launch \   # optionally
+      /path/to/nimrod
+
+The test case provided on Cori has to be ran with a multiple of 11 number of
+ranks - this corresponds to the number of tasks.
+
+### ON CENTOS 7 AND CENTOS STREAM ###
+
+The source code of the test cases can be found in nimall/nimtests/tests. If the
+mknimall.sh was ran with the -t flag, the following scripts should be found in
+the nimall/nimtests directory:
+  * reached-nimtests-all-config.sh
+  * reached-nimtests-all-build.sh
+
+Like above, run the configure script followed by the build script, and refer to
+the respective .txt files to debug.
+
+On the first run of reached-nimtests-all-build.sh or make runtests, all tests
+will fail. This is because NIMROD's tests check for completion and correctness -
+the tests will not only run until they are finished, but also check against a
```

Comment:
`and correctness - the tests` -> `and correctness.  The tests`
...
`finished, but also check a previous` -> `finished.  Also check any previous`

Reply:
I don't agree with the second change, because the tests are checking against previous results to ascertain correctness.